System.CommandLine update

adamsitnik opened the pull request:
This PR consist of few maestro updates required to update the System.CommandLine depedendency:
- arcade: #48453
- roslyn: #48457
- templating: #48467
- sourcelink: #48476
- nuget.client: #48489
- runtime: #48505

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/arcade build 20250414.8

Microsoft.SourceBuild.Intermediate.arcade , Microsoft.DotNet.Arcade.Sdk , Microsoft.DotNet.Build.Tasks.Installers , Microsoft.DotNet.Build.Tasks.Templating , Microsoft.DotNet.Build.Tasks.Workloads , Microsoft.DotNet.Helix.Sdk , Microsoft.DotNet.SignTool , Microsoft.DotNet.XliffTasks , Microsoft.DotNet.XUnitExtensions
 From Version 10.0.0-beta.25212.1 -> To Version 10.0.0-beta.25214.8

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/roslyn build 20250414.5

Microsoft.SourceBuild.Intermediate.roslyn , Microsoft.CodeAnalysis , Microsoft.CodeAnalysis.CSharp , Microsoft.CodeAnalysis.CSharp.CodeStyle , Microsoft.CodeAnalysis.CSharp.Features , Microsoft.CodeAnalysis.CSharp.Workspaces , Microsoft.CodeAnalysis.PublicApiAnalyzers , Microsoft.CodeAnalysis.Workspaces.Common , Microsoft.CodeAnalysis.Workspaces.MSBuild , Microsoft.Net.Compilers.Toolset , Microsoft.Net.Compilers.Toolset.Framework
 From Version 5.0.0-1.25213.1 -> To Version 5.0.0-1.25214.5

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/templating build 20250414.7

Microsoft.SourceBuild.Intermediate.templating , Microsoft.TemplateEngine.Abstractions , Microsoft.TemplateEngine.Authoring.TemplateVerifier , Microsoft.TemplateEngine.Edge , Microsoft.TemplateEngine.Mocks , Microsoft.TemplateEngine.Orchestrator.RunnableProjects , Microsoft.TemplateEngine.TestHelper , Microsoft.TemplateEngine.Utils , Microsoft.TemplateSearch.Common , Microsoft.TemplateSearch.TemplateDiscovery
 From Version 10.0.100-preview.4.25214.2 -> To Version 10.0.100-preview.4.25214.7

adamsitnik committed:
bump System.CommandLine version, remove dependency to System.CommandLine.Rendering (will no longer be supported)

adamsitnik committed:
adjust to breaking changes: the removal of Cli prefix

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/templating build 20250414.8

Microsoft.SourceBuild.Intermediate.templating , Microsoft.TemplateEngine.Abstractions , Microsoft.TemplateEngine.Authoring.TemplateVerifier , Microsoft.TemplateEngine.Edge , Microsoft.TemplateEngine.Mocks , Microsoft.TemplateEngine.Orchestrator.RunnableProjects , Microsoft.TemplateEngine.TestHelper , Microsoft.TemplateEngine.Utils , Microsoft.TemplateSearch.Common , Microsoft.TemplateSearch.TemplateDiscovery
 From Version 10.0.100-preview.4.25214.2 -> To Version 10.0.100-preview.4.25214.8

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/sourcelink build 20250414.2

Microsoft.SourceBuild.Intermediate.sourcelink , Microsoft.Build.Tasks.Git , Microsoft.SourceLink.AzureRepos.Git , Microsoft.SourceLink.Bitbucket.Git , Microsoft.SourceLink.Common , Microsoft.SourceLink.GitHub , Microsoft.SourceLink.GitLab
 From Version 10.0.0-beta.25203.1 -> To Version 10.0.0-beta.25214.2

adamsitnik commented on src/Cli/Microsoft.TemplateEngine.Cli/Commands/create/TemplateCommand.cs:
```diff
@@ -10,10 +10,11 @@
 using Microsoft.TemplateEngine.Edge;
 using Microsoft.TemplateEngine.Edge.Settings;
 using Microsoft.TemplateEngine.Utils;
+using Command = System.CommandLine.Command;
```
I was getting the following error:

```log
'Command' is an ambiguous reference between 'System.CommandLine.Command' and 'Microsoft.DotNet.Cli.Utils.Command'
```

and decided to make it explicit with a `using` rather than using fully qualified name everywhere

adamsitnik commented on src/Cli/dotnet/Extensions/ParseResultExtensions.cs:
```diff
@@ -97,13 +97,13 @@ public static bool IsDotnetBuiltInCommand(this ParseResult parseResult)
     public static bool IsTopLevelDotnetCommand(this ParseResult parseResult)
     {
-        return parseResult.CommandResult.Command.Equals(RootCommand) && string.IsNullOrEmpty(parseResult.RootSubCommandResult());
+        return parseResult.CommandResult.Command.Equals(Microsoft.DotNet.Cli.Parser.RootCommand) && string.IsNullOrEmpty(parseResult.RootSubCommandResult());
```
`CliRootCommand` got renamed to `RootCommand` and it conflicted with `RootCommand` which was imported here with a static using:

https://github.com/dotnet/sdk/blob/afb349499d82e3606cf42ab0bcdc00b3d33557f5/src/Cli/dotnet/Extensions/ParseResultExtensions.cs#L11

carlossanlop commented on src/Compatibility/ApiCompat/Microsoft.DotNet.ApiCompat.Tool/Program.cs:
```diff
@@ -19,94 +19,94 @@ static int Main(string[] args)
             // Important: Keep parameters exposed in sync with the msbuild task frontend.
```
Funnily, My ApiDiff PR is using this newer version of System.CommandLine, and had to only use it in the new tool but not in the existing ones:

https://github.com/dotnet/sdk/pull/46425

I'll soon merge my PR and then there might (or might not) be a merge conflict in this PR, so letting you know in advance. What we (I can do it) need to update in this PR after I merge mine, is remove the exception for the ApiDiff tool so that it instead uses the same version as everyone else.

carlossanlop replied:
On a second thought, I don't want to block you. I'll merge my PR _after_ the prev4 snap, and I'll fix my PR on top of yours.

adamsitnik replied:
@carlossanlop thanks a lot, I really appreciate it!

adamsitnik committed:
Merge branch 'main' into sclUpdate

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/nuget/nuget.client build 6.15.0.7

Microsoft.Build.NuGetSdkResolver , NuGet.Build.Tasks , NuGet.Build.Tasks.Console , NuGet.Build.Tasks.Pack , NuGet.CommandLine.XPlat , NuGet.Commands , NuGet.Common , NuGet.Configuration , NuGet.Credentials , NuGet.DependencyResolver.Core , NuGet.Frameworks , NuGet.LibraryModel , NuGet.Localization , NuGet.Packaging , NuGet.ProjectModel , NuGet.Protocol , NuGet.Versioning
 From Version 6.14.0-preview.1.102 -> To Version 6.15.0-preview.1.7

nkolev92 commented:
I cherry-picked https://github.com/dotnet/sdk/pull/48489 into this PR.

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/runtime build 20250415.12

Microsoft.Bcl.AsyncInterfaces , Microsoft.Extensions.Configuration.Ini , Microsoft.Extensions.DependencyModel , Microsoft.Extensions.FileProviders.Abstractions , Microsoft.Extensions.FileSystemGlobbing , Microsoft.Extensions.Logging , Microsoft.Extensions.Logging.Abstractions , Microsoft.Extensions.Logging.Console , Microsoft.NET.HostModel , Microsoft.NET.ILLink.Tasks , Microsoft.NETCore.App.Host.win-x64 , Microsoft.NETCore.App.Ref , Microsoft.NETCore.App.Runtime.win-x64 , Microsoft.NETCore.Platforms , Microsoft.Win32.SystemEvents , System.CodeDom , System.Composition.AttributedModel , System.Composition.Convention , System.Composition.Hosting , System.Composition.Runtime , System.Composition.TypedParts , System.Configuration.ConfigurationManager , System.Formats.Asn1 , System.IO.Hashing , System.Reflection.MetadataLoadContext , System.Resources.Extensions , System.Security.Cryptography.Pkcs , System.Security.Cryptography.ProtectedData , System.Security.Cryptography.Xml , System.Security.Permissions , System.ServiceProcess.ServiceController , System.Text.Encoding.CodePages , System.Text.Json , System.Windows.Extensions , VS.Redist.Common.NetCore.SharedFramework.x64.10.0 , VS.Redist.Common.NetCore.TargetingPack.x64.10.0 , Microsoft.SourceBuild.Intermediate.runtime.linux-x64
 From Version 10.0.0-preview.4.25211.19 -> To Version 10.0.0-preview.4.25215.12

ViktorHofer commented:
@premun we need to flow command-line-api into the VMR from sdk for this PR. Can we change the flow back temporarily to unblock the PR?

ViktorHofer committed:
Temporarily synchronize sourcelink again

ViktorHofer committed:
Delete src/SourceBuild/patches/runtime/0001-treat-targetos-and-rid-as-local-properties.patch

premun commented:
@ViktorHofer do you mean to enable to old kind of synchronization for sourcelink?
Why can't we make the change sourcelink directly?

ViktorHofer commented:
The change was already made to sourcelink but everything needs to flow-in together via sdk in this case.

premun commented:
@ViktorHofer then please change the `disableSynchronization` for sourcelink in this PR to see if it works.
(and update the V.D.xml record of sourcelink)

premun commented:
Ah, I see. Then yeah, feel free to merge this

ViktorHofer commented:
@adamsitnik I think you need to revert changes under src/SourceBuild.

adamsitnik committed:
revert changes under src/SourceBuild

akoeplinger committed:
Add runtime patch

adamsitnik commented:
I've re-run the failed sdk test leg as the failing test has 6% failure rate.

@ViktorHofer how can we overcome the source build failure?

adamsitnik committed:
Revert "Update dependencies from https://github.com/dotnet/arcade build 20250414.8"

This reverts commit c2a6e14b29d302b66b4e7b018c33627c92090880.

ViktorHofer committed:
Merge branch 'main' into sclUpdate

ViktorHofer commented:
Ok that worked. The latest update should resolve the test failure in the source build job.

ViktorHofer commented:
We will need to temporarily break source-build stage2 for this. Merge the PR, wait for a new official build and then re-bootstrap.

adamsitnik commented:
@ViktorHofer the sdk-source-build has failed with:

```log
/__w/1/vmr/eng/tools/BinaryToolKit/Program.cs(61,20): error CS0246: The type or namespace name 'CliCommand' could not be found (are you missing a using directive or an assembly reference?) [/__w/1/vmr/eng/tools/BinaryToolKit/BinaryToolKit.csproj]
/__w/1/vmr/eng/tools/BinaryToolKit/Program.cs(71,42): error CS0246: The type or namespace name 'CliCommand' could not be found (are you missing a using directive or an assembly reference?) [/__w/1/vmr/eng/tools/BinaryToolKit/BinaryToolKit.csproj]
/__w/1/vmr/eng/tools/BinaryToolKit/Program.cs(12,28): error CS0246: The type or namespace name 'CliArgument<>' could not be found (are you missing a using directive or an assembly reference?) [/__w/1/vmr/eng/tools/BinaryToolKit/BinaryToolKit.csproj]
/__w/1/vmr/eng/tools/BinaryToolKit/Program.cs(18,28): error CS0246: The type or namespace name 'CliOption<>' could not be found (are you missing a using directive or an assembly reference?) [/__w/1/vmr/eng/tools/BinaryToolKit/BinaryToolKit.csproj]
/__w/1/vmr/eng/tools/BinaryToolKit/Program.cs(25,28): error CS0246: The type or namespace name 'CliOption<>' could not be found (are you missing a using directive or an assembly reference?) [/__w/1/vmr/eng/tools/BinaryToolKit/BinaryToolKit.csproj]
/__w/1/vmr/eng/tools/BinaryToolKit/Program.cs(33,28): error CS0246: The type or namespace name 'CliOption<>' could not be found (are you missing a using directive or an assembly reference?) [/__w/1/vmr/eng/tools/BinaryToolKit/BinaryToolKit.csproj]
```

Does it mean I should actually port everything in https://github.com/dotnet/sdk/blob/main/src/SourceBuild/content/eng/tools/BinaryToolKit ? (I did that initially and then you told me to revert it).

ViktorHofer commented:
See my previous comment. We can't fix source-build stage2 here as then source-build stage1 would be broken. We will need to force-merge that PR and then unbreak it by doing some work. I will drive that today.
